### PR TITLE
Implemented `knife user key delete` and `knife client key delete`.

### PR DESCRIPTION
**Pull request by tylercloke**
Ping @chef/lob 

@danielsdeleo @jdmundrawala @btm /cc client-core

### CODE REVIEW

**Comment by danielsdeleo** [comment]
@chef/client-core

File: lib/chef/knife/key_delete.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  class Knife
+    # Service class for UserKeyDelete and ClientKeyDelete, used to delete keys.
+    # Implements common functionality of knife [user | org client] key delete.
+    #
+    # @author Tyler Cloke
+    #
+    # @attr_accessor [Hash] cli input, see UserKeyDelete and ClientKeyDelete for what could populate it
+    class KeyDelete
+
+      attr_accessor :config
```
Comment (btm) [inline lib/chef/knife/key_delete.rb]: Is this used in the tests?

Comment (tylercloke) [inline lib/chef/knife/key_delete.rb]: In this case I don't need it. I'll remove it.

Comment (tylercloke) [inline lib/chef/knife/key_delete.rb]: Fixed.

File: lib/chef/knife/key_delete.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    # @attr_accessor [Hash] cli input, see UserKeyDelete and ClientKeyDelete for what could populate it
+    class KeyDelete
+      def initialize(name, actor, actor_field_name, ui)
+        @name = name
+        @actor = actor
+        @actor_field_name = actor_field_name
+        @ui = ui
+      end
+
+      def confirm
```
Comment (jdmundrawala) [inline lib/chef/knife/key_delete.rb]: can you call this `confirm!`

**Comment by jdmundrawala** [comment]
LGTM :+1:

File: lib/chef/knife/key_delete.rb (hunk 2 above)
Comment (tylercloke) [inline lib/chef/knife/key_delete.rb]: OK!

**Comment by btm** [comment]
:+1:

**Pull request merged**